packages/fxa-settings/src/models/Account.ts

```diff
@@ -569,27 +577,43 @@ export class Account implements AccountData {
     service?: string,
     redirectTo?: string
   ): Promise<PasswordForgotSendCodePayload> {
-    let options: {
-      service?: string;
-      resume?: string;
-      redirectTo?: string;
-    } = {
-      resume: 'e30=', // base64 json for {}
-    };
-
-    // Important! Only set service when it's Firefox Sync
-    if (service && service === MozServices.FirefoxSync) {
-      options.service = 'sync';
-    } else {
-      options.service = service;
-    }
+    try {
+      if (service && service === MozServices.FirefoxSync) {
+        service = 'sync';
+      }
 
-    if (redirectTo) {
-      options.redirectTo = redirectTo;
+      const result = await this.apolloClient.mutate({
+        mutation: PASSWORD_FORGOT_SEND_CODE,
+        variables: {
+          input: {
+            email,
+            // Only include the `service` option if the service is Sync.
+            // This becomes a query param (service=sync) on the email link.
+            // We need to modify this in FXA-7657 to send the `client_id` param
```

Comment:
Looking back at the code prior to the [PR](https://github.com/mozilla/fxa/pull/15505) for FXA-6858, I see the logic as described in the comment, which is where the comment originated from. However, looking at the state after FXA-6858 the service _would_ always get passed. Either way the code doesn't match the comment, so glad you called this out!
 
I agree that the service shouldn't be be relayed with a client_id as its value, so I will go with the state prior to FXA-6858, which is probably the correct way to do this.

Reply:
@dschom Going to reiterate a couple things I just said in Slack but I noticed a couple of things and want to leave a note here for future reference.

With the most recent push, when going through the OAuth flow, the product name gets the default "account settings" since the URL [doesn't contain a `client_id` or `service` param to look up client info from](https://github.com/mozilla/fxa/blob/7dd2de5841a1ca71f0f17be14b04336cf5183d0e/packages/fxa-settings/src/models/integrations/oauth-integration.ts#L201-L202). I was going to suggest this change which fixes that problem...

```
  ...
  resume: 'e30=',
  // NOTE: The 'service' param should only ever be 'sync' or a Sync client ID.
  // However, we're passing along the 'client_id' here as 'service' so when users
  // open the reset PW link, 'service' will contain the client ID and we can
  // properly look up and display the associated RP name. We would fix this but
  // we'll be ripping out this functionality in FXA-6670.
  service: service === MozServices.FirefoxSync ? 'sync' : service,
},
...
```
... However I noticed that [we aren't even checking for the Sync Desktop case](https://github.com/mozilla/fxa/blob/7dd2de5841a1ca71f0f17be14b04336cf5183d0e/packages/fxa-settings/src/pages/ResetPassword/index.tsx#L115) when we call this function. Testing the Sync flow with my suggestion gets me here:
![image](https://github.com/mozilla/fxa/assets/13018240/b9d766cf-6a59-4bd5-b320-93ef20feaba2)

I tried adding `integration.getServiceName()` to that function call, which does appear to work, but we're taken to this screen:
![image](https://github.com/mozilla/fxa/assets/13018240/800e2818-bc10-421b-8b10-44b1725dfc84)

and I'm not sure if that screen is correct now since users aren't signed in after resetting a PW.

Either way this is an existing issue. I just verified on `main` that resetting a PW through Sync shows "You’re now ready to use account settings". So I think we want the change I suggested here to fix it for the OAuth flow, and maybe file an issue for the service name in Sync and put it into that clear messaging epic (I can do this, we'll probably cancel it since we're moving to codes but it'll be good to document).

Reply:
Okay, went back to supplying service. And retested

123Done End Page (on local):
![image](https://github.com/mozilla/fxa/assets/94418270/a4038c39-f5ef-414e-8d17-158d73a98f0d)

Sync End Page (on local):
![image](https://github.com/mozilla/fxa/assets/94418270/c111569b-a877-4c09-81de-e24994b0ff31)

Sync End Page (on production):
![image](https://github.com/mozilla/fxa/assets/94418270/5b0e151c-cab0-4be9-9572-79f9aa429271)


In my opinion, even though sync isn't perfect. It's as good or better now.

Reply:
Filed: https://mozilla-hub.atlassian.net/browse/FXA-9103